test/pylib/scylla_cluster.py

```diff
@@ -1249,6 +1278,21 @@ async def _after_test(self, _request) -> str:
             self.current_test_case_full_name = ''
         self.is_after_test_ok = True
         cluster_str = str(self.cluster)
+
+        # cancel all other tasks in ScyllaClusterManager
+        self.logger.info(self.repr_tasks_history())
+        for key, value in self.tasks_history.items():
+            if "after-test" not in value.path_qs:
+                key.cancel(msg="test finished")
+        for key, value in self.tasks_history.items():
+            if not key.done() and "after-test" not in value.path_qs:
+                self.logger.info("wait for cancelling task:%s, request:%s", key.get_name(), value.path_qs)
+                try:
+                    await key
```

Comment:
I wonder if we should introduce a timeout here and kill everything if it doesn't finish after timeout.

(But we do need to make sure to cleanup resources -- I don't know if some of these tasks might be responsible for that)

Reply:
Asyncio tasks cannot be killed, only cancelled, even cancel is now guaranteed that the task is finished immediately.

already wrote about it here https://github.com/scylladb/scylladb/issues/16472#issuecomment-1898807664

Reply:
regarding cleanup and timeout - will do additional testing

Reply:
added timeout and waited for PUT actions in no raise an error

regarding cleanup. 
as I mentioned is https://github.com/scylladb/scylladb/issues/16472#issuecomment-1898807664 
Asyncio tasks cannot be killed, only finished